tools/minify.cpp

```diff
@@ -5,74 +5,43 @@
-char* exe_name;
+cxxopts::Options options("minify", "Runs the parser against the given json files in a loop, measuring speed and other statistics.");
 
-void print_usage(std::ostream& out) {
-  out << "Usage: " << exe_name << "  [-a ARCH] <jsonfile>" << std::endl;
-  out << std::endl;
-  out << "Runs the parser against the given json files in a loop, measuring speed and other statistics." << std::endl;
-  out << std::endl;
-  out << "Options:" << std::endl;
-  out << std::endl;
-  out << "-a IMPL      - Use the given parser implementation. By default, detects the most advanced" << std::endl;
-  out << "               implementation supported on the host machine." << std::endl;
+void exit_usage(std::string message) {
+  std::cerr << message << std::endl;
+  std::cerr << options.help() << std::endl;
+  std::cerr << "Available parser implementations:" << std::endl;
   for (auto impl : simdjson::available_implementations) {
-    out << "-a " << std::left << std::setw(9) << impl->name() << " - Use the " << impl->description() << " parser implementation." << std::endl;
+    std::cerr << "-a " << std::left << std::setw(9) << impl->name() << " - Use the " << impl->description() << " parser implementation." << std::endl;
   }
+  exit(1);
 }
 
-void exit_usage(std::string message) {
-  std::cerr << message << std::endl;
-  std::cerr << std::endl;
-  print_usage(std::cerr);
-  exit(EXIT_FAILURE);
-}
+int main(int argc, char *argv[]) {
+  options.add_options()
+    ("a,arch", "Parser implementation (by default, detects the most advanced implementation supported on the host machine).", cxxopts::value<std::string>())
+    ("f,file", "File name.", cxxopts::value<std::string>())
+    ("h,help", "Print usage.")
+  ;
 
+  options.parse_positional({"file"});
+  auto result = options.parse(argc, argv);
 
-struct option_struct {
-  char* filename{};
- 
-  option_struct(int argc, char **argv) {
-    int c;
+  if(result.count("help")) {
+    exit_usage("");
+  }
 
-    while ((c = getopt(argc, argv, "a:")) != -1) {
-      switch (c) {
-      case 'a': {
-        const simdjson::implementation *impl = simdjson::available_implementations[optarg];
-        if (!impl) {
-          std::string exit_message = std::string("Unsupported option value -a ") + optarg + ": expected -a  with one of ";
-          for (auto imple : simdjson::available_implementations) {
-            exit_message += imple->name();
-            exit_message += " ";
-          }
-          exit_usage(exit_message);
-        }
-        simdjson::active_implementation = impl;
-        break;
-      }
-      default:
-        // reaching here means an argument was given to getopt() which did not have a case label
-        exit_usage("Unexpected argument - missing case for option "+
-                    std::string(1,static_cast<char>(c))+
-                    " (programming error)");
-      }
-    }
+  if(!result.count("arch")) exit_usage("No implementation specified.");
```

Comment:
I think that "arch" should be an optional argument, as it is in the other tools.

Reply:
Make "arch" an optional argument. Do not touch simdjson::active_implementation when "arch" is not specified.